p2p/src/authenticated/actors/tracker/reservation.rs

```diff
@@ -36,15 +39,16 @@ impl<E: Spawner + Metrics, P: Array> Reservation<E, P> {
 
 impl<E: Spawner + Metrics, P: Array> Drop for Reservation<E, P> {
     fn drop(&mut self) {
+        let mut closer = self.closer.take().expect("Reservation::drop called twice");
+
         // If the mailbox is not full, we can release the reservation immediately without spawning a task.
-        let Err(e) = self.closer.try_send(self.metadata.clone()) else {
+        let Err(e) = closer.try_send(self.metadata.clone()) else {
             // Sent successfully, nothing to do.
             return;
         };
         if e.is_full() {
             // If the mailbox is full, we need to spawn a task to handle the release. If we used `block_on` here,
             // it could cause a deadlock.
-            let mut closer = self.closer.clone();
             let metadata = self.metadata.clone();
```

Comment:
I suppose we can do the same with `Metadata`?

Reply:
Not as useful since you still need to clone it once to put it in `try_send` since that function takes ownership of it and you may need to use it in the later call to `send`. I would consider this the hot path since the use of `send` is probably less common (though I guess you do save a single `clone()` call in this "worst case").

Also we would require cloning in the `fn metadata` function since it has to return a new value rather than a reference.

So I would suggest not doing `Option` for `Metadata`